### PR TITLE
Add Deno support

### PR DESCRIPTION
**Pull request by JumpLink**
Add Deno support to be able to use this package in Node.js and Deno projects.

* [Upgrade TypeScript to 5.3.3](https://github.com/beabee-communityrm/beabee-common/pull/2/commits/f9b35302bf2efc17fb39ae983cbfa12603633476)
* [Add mod.ts which can be used from deno](https://github.com/beabee-communityrm/beabee-common/pull/2/commits/db2a4231f3a919fa55138af703d0a932ac8bed59)
* Changed the codebase to work with Deno and Node.js
* Use [esbuild](https://esbuild.github.io/) to build for ESM and CJS
* Changed package type to ESM
* Portend and replaced Jest tests to Deno Test
* Portend and replaced Prettier with Deno Formatter
* Added Github action workflows to run Deno tests and check Node.js ESM and CJS builds
* Upgrade all dependencies to latest
* Moved types to the types folder to be consistent with the other code bases and to make it easier to move types to common

## Breaking changes

The folder structure has changed, but if only the index was imported, no adjustments should be necessary

### CODE REVIEW

**Review by wpf500 — COMMENTED** [review]
Ah nice one, good to get this library updated etc.

Do we really need the tests in Deno and Node? I guess we could just keep one?

**Comment by JumpLink** [comment]
> Ah nice one, good to get this library updated etc.
> 
> Do we really need the tests in Deno and Node? I guess we could just keep one?

Yes we can just keep one, in this case I would keep the Deno tests because testing is a feature of Deno and we could save jest as an additional dependency. Then I just write a simple test that checks if the types for Node.js are correct to make sure that the same dependencies are used

**Comment by wpf500** [comment]
Sounds good to me :+1:

**Comment by JumpLink** [comment]
if we use Deno for the tests we can also use the formatter of Deno to save another dependency to prettier

File: src/types/rule-pperator-params.ts
Comment (wpf500) [inline src/types/rule-pperator-params.ts:1]: Typo in file name, should be `rule-operator-param.ts`

**Review by wpf500 — CHANGES_REQUESTED** [review]
Thanks for doing so much clearing up of my tangled code! I'd really like to do some more thinking about how to structure this library at some point but this is a great start! :tada:

**Comment by JumpLink** [comment]
@wpf500 Thank you! I'm now done with my changes in the PR :)

**Pull request merged**